<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8">
    <title> RNApredator bacterial sRNA target prediction Webserver - Help - http://www.tbi.univie.ac.at/ </title>
     <style type="text/css">
    table {table-layout:fixed; text-align:center; overflow:auto; border:0px black solid; width:95%;}
    table.borders {table-layout:fixed; text-align:center; overflow:auto; border:1px black solid; width:95%;}
    table td {text-align:center; border:0px black solid; }
    .tooltip2 {
    position: absolute;
    display: none;
    background-color: #FFFFFF;
    border-width: 1px;
    border-style: solid;
    border-color: #000000;
    padding: 2px;
    }
    .tooltip2 .head{
    background-color: #006666;
    color: white;
    padding: 2px;
    }
    .tooltip2 .content{
    font-size: 12px;
    }
    .tooltip2 .content td{
    width: 50%;
    font-size: 10px;
    text-align: center;
    }
  </style>
  </head>
  <body>
    <div style="text-align:center">
      <table style="width:100%;">	
	<tr>
	  <td rowspan="2" style="width:45%;"><img src="./pictures/banner_final.png" alt="Banner"></td>
          <td style="width:11%;"><a href="introduction.html"><img src="./pictures/introduction.png" alt="Introduction"></a></td>
	  <td style="width:18%;"><a href="available_genomes.cgi"><img src="./pictures/available_genomes.png" alt="Available Genomes"></a></td>
          <td><a href="target_search.cgi"><img src="./pictures/target_search.png" alt="Target Search"></a></td>
	  <td><a href="help.html"><img src="./pictures/help.png" alt="Help"></a></td>
	</tr>
	<tr>
	  <td></td>
	  <td colspan="2" style="background-color:#dcdcdc;height:75%;"><a href="introduction.html#history"> History:</a> Version 1.33 online</td>
	  <td></td>
	</tr>
      </table>
      <hr>
      </div>
    <div style="text-align:center">
      <h3>= Help =</h3>	
    </div>
    <div style="text-align:left">
      <p style="margin-top: 0.17in;"><strong>Introduction</strong></p>
      <p>The RNApredator software facilitates the prediction of sRNA-mRNA
	interactions in bacteria. Currently targets can be searched in more
	than 1300 bacterial species. RNApredator uses a dynamic programming
	approach to compute putative targets. The main machinery used by
	RNApredator to find targets is the program called <a href="http://www.bioinf.uni-leipzig.de/Software/RNAplex/">RNAplex (Version 1.8.4)</a>.
	Post-processing steps allow to look at accessibility differences on
	the target mRNA before and after the binding of the sRNAs. It is
	further possible to make a Gene Ontology(GO) and pathways (KEGG) term
	enrichment analysis on a set of selected targets.</p>
      <p style="margin-top: 0.17in;"><strong>Input</strong></p>
      <p>The webserver takes as input a genome/plasmid from a set of 2155
	sequences. This can either be entered through the corresponding
	NCBI-Accession Number or the NCBI-Taxonomy ID (Illustration 1). The
	desired genome can also be searched in a tree containing all
	available genomes and grouped into
	phylum/class/order/family/genus/species (Illustration 2). Once the
	desired genome has been selected, a sRNA sequence should be entered,
	without line breaks and without header. The target search is launched
	after the predict button has been pressed. 
      </p>
      <p>
	Enabling the "include suboptimal interactions" checkbox adds suboptimal results to the result list.
      </p>
      <p>An alternative way to enter the genome of interest is to choose it
	from a tree containing all available
	genomes and grouped into
	phylum/class/order/family/genus/species.
      </p>
      <img src="pictures/inputdemo.png" alt="inputdemo"><br>
      
	<I>Illustration1:<br>
	  Input field of RNApredator. l.h.s Genomes can be introduced<br>
	  either through a NCBI-Accession Number or NCBI-Taxonomy ID. r.h.s<br>
	  The sRNA sequence, without header can be pasted in the corresponding<br>
	  field</i>
      <br>
      <br>
      
      <img src="pictures/tree.png" alt="inputdemo"><br>
      
	<I>Illustration2:<br>
	  Tree representation of the genomes available in RNApredator. The<br>
	  user can browse by clicking on the arrow symbols. Genomes/plasmids<br>
	  can be selected for target search by clicking on the node designation</i>
      <br>
      
      <p style="margin-top: 0.17in;"><strong>Processing</strong></p>
      <p style="margin-bottom: 0in">The results are calculated as follows:<br>
	RNAplex is called with the parameter interaction length set to 30 and precomputed accessibilities produced with RNAplfold (Version 1.8.4)
	for the mRNAs and RNAup (Version 1.8.4) for the sRNAs.
	If suboptimal interactions were requested additionally an energy-threshold of -8 and a duplex distance of 20 is used running RNAplex. 
	The <a href="http://www.tbi.univie.ac.at/~ronny/RNA/RNAplex.html">RNAplex man page</a> gives a detailed explanation of these parameters.
      </p>
      <p style="margin-bottom: 0in">
	RNAplfold calculation of the mRNA accessiblities was performed with window-size (-W) set to 240, span (-L) to 160, length of the unstructured region to 30, requesting opening energies (-O) (see <a href="http://www.tbi.univie.ac.at/~ronny/RNA/RNAplfold.html">RNAplfold man-page</a> for details).
	RNAup computation of sRNA accessibility is called with the parameter for length of the unstructured region (-u) set to 40 or the sequence lenght if shorter
	(see <a href="http://www.tbi.univie.ac.at/~ronny/RNA/RNAup.html">RNAup man-page</a> for details).
      </p>
      
      <p style="margin-top: 0.17in;"><strong>Output</strong></p>
      <p style="margin-bottom: 0in">The webserver outputs a list of
	sRNA-mRNA interactions (including suboptimal interactions if enabled in input), which
	is sorted based on the energy of
	interaction returned by the
	dynamic programming approach. It is
	possible to sort the results
	based on the energy/position/name of
	target/Z-score/Genomic
	coordinate/Accession. Further the output can
	be filtered based on the
	position of the targets relative to the
	start codon. Finally it is also
	possible to  select only the
	25/50/75/100 best interactions. It should
	be noted that all the
	results can be downloaded in tabular format.<br>
	You can view a example result table of the sample input
	<a href="example.html">here (E.coli str. K12 substr. MG1655 - MicA sRNA )</a>.
	
      </p>
      <p style="margin-bottom: 0in">It is also possible to select a set of
	interations of interest and post-process them. The post-processing
	step will return a list of significantly enriched GO/KEGG terms as
	well as the variation in accessibility around the ribosomal target
	site before and after the binding of the sRNA.</p>
      <p style="margin-bottom: 0in">
	Due to the large amount of output data, job results are only kept for <strong>60 days</strong>
	after computation! 
      </p>
    </div>
    
 <div class="copyright">
	<hr>
	2010 - 2012 TBI - <a href="mailto:egg@tbi.univie.ac.at">Contact</a> -  <a href="http://nar.oxfordjournals.org/content/early/2011/06/14/nar.gkr467.full">Publication</a>
   </div>
  </body>
</html>


