<!DOCTYPE html>
<html>
  <head>
    <meta http-equiv="content-type" content="text/html; charset=ISO-8859-1">
    <title> RNApredator bacterial sRNA target prediction Webserver - Introduction </title>
    <style type="text/css">
      table {black solid; }
      table thead {background:#ccc; }
      table td.last { padding-right:15px; }
      .scroll {height:20ex; overflow-y:scroll; overflow-x:hidden; }
      table td {text-align:center; border:0px black solid; }
    </style>
  </head>
  <body>
    <div style="text-align:center">
      <table width="100%">	
	<tbody><tr>
	    <td rowspan="2" height="100%" width="45%"><img src="pictures/banner_final.png"></td>
            <td width="11%"><a href="introduction.html"><img src="pictures/introduction.png" alt="Introduction" border="0"></a></td>
	    <td width="18%"><a href="available_genomes.cgi"><img src="pictures/available_genomes.png" alt="Available Genomes" border="0"></a></td>
            <td><a href="target_search.cgi"><img src="pictures/target_search.png" alt="Target Search" border="0"></a></td>
	    <td><a href="help.html"><img src="pictures/help.png" alt="Help" border="0"></a></td>
	  </tr>
	  <tr>
	    <td></td>
	    <td colspan="2" bgcolor="#dcdcdc" height="75%"><a href="http://localhost:800/RNApredator/introduction.html#history"> History:</a> Version 1.10 online</td>
	    <td></td>
	  </tr>
      </tbody></table>
      <hr>
    </div>
    <div style="text-align:center">
      <h3 id="stepx_title">= Introduction =</h3>
    </div>
    <div style="text-align:left">
      <p style="margin-top: 0.17in; page-break-after: avoid"><font face="Albany, sans-serif"><font size=4>Introduction</font></font></p>
      <p>sRNAs are non-coding RNAs found in bacteria. sRNAs are very
	heterogeneous both in sizes, structures and functions Most
	of them act as post-transcriptional regulators by interacting
	with the 5' untranslated region of mRNA transcripts, modifying
	their stability and/or their ability to be translated. 
      </p>
      <p>The RNApredator software facilitates the prediction of sRNA-mRNA
	interactions in bacteria. Currently targets can be searched in more
	than 1300 bacterial species. RNApredator uses a dynamic programming
	approach to compute putative targets. The main machinery used by
	RNApredator to find targets is the program called  <a href="http://www.bioinf.uni-leipzig.de/Software/RNAplex/">RNAplex</a>.
	RNAplex is a new approach for RNA-RNA interaction search, which has
	a prediction accuracy  similar to that of algorithms that explicitly consider intramolecular
	structures, but running at least three orders of magnitude faster than RNAup and IntaRNA. This is achieved
	by using a combination of pre-computed accessibility profiles, with an approximate energy-
	model, allowing to mimic the effects of the competition between intra- and intermolecular interactions.
	The addition of the accessibility information into the computation of the sRNA-mRNA interactions allows to improve the specificity as very
	structured targets are discarded.</p>

      <p style="margin-top: 0.08in; margin-bottom: 0.08in">
	<img src="pictures/RNAplex.png" name="graphics1" align=left width=30% border=0>
	<br clear=left><font size=3><i>	
	    Illustration1:<br>
	    Top: In algorithm like TargetRNA or RNAhybrid, the energy gained<br>
	    by adding a base-pair (i,j) to an existing duplex (S(i,j;i-1,j+)<br>
	    does not take the energy necessary to free base-pairs i and j <br>
	    from all the intramolecular interactions they are involved in. <br>
	    Bottom:
	    For each nucleotide added to the duplex, the energy necessary<br>
	    to remove the nucleotide from all its intramolecular interactions (d<sub>1</sub><sup>x</sup>,
	    d<sub>1</sub><sup>y</sup>)<br> is taken into account. This allows to
	    reach a greater specificity in the target predictions.</i></font>
      </p>
      <p>Besides the raw prediction of binding partners, RNApredator allows
	the analysis of the results in different post-processing steps. It is
	possible to look at enriched GO terms/pathways in a set of
	user-selected targets. Furthermore RNApredator offers the user the
	possibility to look at the difference of accessibility around the
	ribosome binding site before and after binding of the sRNA to its
	target, which is useful when the user wants to know the sRNA will up
	or downregulate its target.</p>
    </div>
    <hr>
    <div style="text-align:center">
      <h3><a name="history">= History =</a></h3>
      <table border="1" width="100%">
	<tr>
	  <td><b>Version</b></td>
	  <td><b>Git-commit</b></td>
	  <td><b>Date</b></td>
	  <td><b>Description</b></td>
	</tr>
	<tr>
	  <td>1.00</td>
	  <td>1</td>
	  <td>Date</td>
	  <td>Release Version</td>
      </tr>
	<tr>
	  <td>1.10</td>
	  <td>17</td>
	  <td>12.03.2012 </td>
	  <td>Fixed several cases of broken postprocessing, enhanced arrangment of navigation elements<br>
	    unified progress-tracking during calculation and added history</td>
	</tr>
	<tr>
	  <td>1.25</td>
	  <td>25</td>
	  <td>29.04.2012 </td>
	  <td>Added checkbox to switch suboptimals on/off<br>
	    Changes to the repository, that make future updates simpler, e.g. automatic configuration depending on hostname<br>
	    Added better accounting of usage statistics and a link to the publication</td>
	</tr>	
    </table> 
    </div>
    <div class="copyright">
      <hr>
      2010-2012 TBI - <a href="mailto:egg@tbi.univie.ac.at">Contact</a> 
    </div>
  </body>
</html>


